src/runtime_src/hip/core/event.cpp

```diff
@@ -4,66 +4,180 @@
 #include "event.h"
 
 namespace xrt::core::hip {
+event::
+event(std::shared_ptr<stream>&& s)
+  : command(std::move(s))
+{
+  ctype = type::event;
+}
+
+void
+event::
+record(std::shared_ptr<stream> s)
+{
+  cstream = std::move(s);
+  auto ev = std::dynamic_pointer_cast<event>(command_cache.get(static_cast<command_handle>(this)));
+  throw_invalid_handle_if(!ev, "event passed is invalid");
+  if (is_recorded()) {
+    // already recorded
+    //cstream->erase_cmd(ev); It is commented will be uncomment after stream PR is pushed
+  }
+  // update recorded commands list
+  //cstream->enqueue_event(std::move(ev));It is commented will be uncomment after stream PR is pushed
+  set_state(state::recorded);
+}
 
 bool
-command::
-submit()
+event::
+is_recorded()
 {
-    return true; //temporary
+  return get_state() >= command::state::recorded;
 }
 
 bool
-command::
-wait()
+event::
+query()
 {
-    return true; //temporary
+  for (auto it = recorded_commands.begin(); it != recorded_commands.end(); it++){
+    state command_state = (*it)->get_state();
+    if (command_state != state::completed){
+      return false;
+    }
+  }
+  return true;
 }
 
 bool
 event::
-submit()
+synchronize()
 {
-    return true; //temporary
+  for (auto it = recorded_commands.begin(); it != recorded_commands.end(); it++){
+    state command_state = (*it)->get_state();
+    if (command_state < state::completed){
+      (*it)->wait();
+      (*it)->set_state(state::completed);
+    }
+  }
+  set_state(state::completed);
+  for (auto it = chain_of_commands.begin(); it != chain_of_commands.end(); it++){
+    (*it)->submit(true);
+  }
+  return true;
 }
 
 bool
 event::
 wait()
 {
-    return true; //temporary
+  ctime = std::chrono::system_clock::now();
+  return synchronize();
+}
+
+bool
+event::
+submit(bool)
+{
+  return true;
+}
+
+std::shared_ptr<stream>
+event::
+get_stream()
+{
```

Comment:
I wonder about the current coding rules to be so sparse instead of this being just 1 line...
At the end it is difficult to have a big picture with so much space wasted on the page.
Or perhaps this is to be sure to make the code hard to read and understand? ;-)

Reply:
This is my coding style, and I am adamant about it :-)
I want all interesting stuff on the left side of the page, which makes scanning code much much easier.
Get a bigger monitor if you losing the big picture.

Reply:
Even with a bigger monitor I could still say that I could grasp even more content with a more compact coding style. ;-)

Reply:
Yeah, you could write everything on one line, who needs whitespaces.